### ISSUE TITLE
[Accessibility] Localization Issue in Search Form Component

### ISSUE THREAD

**Issue by kivuvarosekivuvan** [ushahidi/platform#4980]
# Summary

## Describe the bug
The strings in the `search-form` component are not localized. Specifically, the strings "Filters and sorting" and "No saved filters. Saved filter(s) will appear here for you to select once made available." are not being translated or provided through a localization mechanism.

## To Reproduce
1. Navigate to the `web-mzima-client/src/app/shared/components/search-form/search-form.component.html` file.
2. Observe the following code on line 240 and 325 respectively:

```html
240: <h3>Filters and sorting</h3>
325: <span class="no-saved-filters" postfix *ngIf="savedSearches?.length === 0">
  No saved filters. Saved filter(s) will appear here for you to select once made available.
</span>
```

## Expected behavior
The strings in the `search-form` component should be localized and provided through a translation or localization mechanism, allowing the application to support multiple languages.

## How to solve
To solve this issue, we will:

1. Identify all the strings in the `search-form` component that need to be localized.

2. Ensure that the localization mechanism is properly set up and configured in the application to handle the localization of the `search-form` component.

## Screenshots
<img width="1200" alt="Screenshot 2024-07-24 at 09 20 10" src="https://github.com/user-attachments/assets/18392676-97a6-46d4-8180-cd1931bf47f1">
<img width="1200" alt="Screenshot 2024-07-24 at 09 20 34" src="https://github.com/user-attachments/assets/3e0dd069-be33-4b1d-886d-e0b36d654c47">
